# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManagerService.cpp
# Commit: a604c144191f
# Full Hash: a604c144191f5c3c65169c072d4e52a1e2c1c23f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 09:46:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Change nsIQuotaManagerService::ClearStoragesForPrincipal to support clearing of entire origins only; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195505
# ==============================================================================

diff -r 281a926003fd -r a604c144191f dom/quota/QuotaManagerService.cpp
--- a/dom/quota/QuotaManagerService.cpp	Wed Oct 09 05:34:36 2024 +0000
+++ b/dom/quota/QuotaManagerService.cpp	Wed Oct 09 05:34:37 2024 +0000
@@ -1021,7 +1021,7 @@
 NS_IMETHODIMP
 QuotaManagerService::ClearStoragesForPrincipal(
     nsIPrincipal* aPrincipal, const nsACString& aPersistenceType,
-    const nsAString& aClientType, nsIQuotaRequest** _retval) {
+    nsIQuotaRequest** _retval) {
   MOZ_ASSERT(NS_IsMainThread());
   MOZ_ASSERT(aPrincipal);
 
@@ -1055,24 +1055,9 @@
         return principalInfo;
       }()));
 
-  QM_TRY_INSPECT(const auto& clientType,
-                 ([&aClientType]() -> Result<Maybe<Client::Type>, nsresult> {
-                   if (aClientType.IsVoid()) {
-                     return Maybe<Client::Type>();
-                   }
-
-                   Client::Type clientType;
-                   QM_TRY(MOZ_TO_RESULT(Client::TypeFromText(
-                              aClientType, clientType, fallible)),
-                          Err(NS_ERROR_INVALID_ARG));
-
-                   return Some(clientType);
-                 }()));
-
   RefPtr<Request> request = new Request();
 
-  mBackgroundActor
-      ->SendClearStoragesForOrigin(persistenceType, principalInfo, clientType)
+  mBackgroundActor->SendClearStoragesForOrigin(persistenceType, principalInfo)
       ->Then(GetCurrentSerialEventTarget(), __func__,
              BoolResponsePromiseResolveOrRejectCallback(request));
 